source/slang/ir-glsl-legalize.cpp

```diff
@@ -376,27 +434,46 @@ GLSLSystemValueInfo* getGLSLSystemValueInfo(
         }
 
         name = "gl_Layer";
-        requiredType = context->getBuilder()->getBasicType(BaseType::Int);
+        requiredType = builder->getBasicType(BaseType::Int);
     }
     else if (semanticName == "sv_sampleindex")
     {
+        // uint in hlsl, int in glsl
+        // https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl_SampleID.xhtml
+
+        requiredType = builder->getBasicType(BaseType::Int);
         name = "gl_SampleID";
     }
     else if (semanticName == "sv_stencilref")
     {
+        // uint in hlsl, int in glsl
+        // https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_shader_stencil_export.txt
+
+        requiredType = builder->getBasicType(BaseType::Int);
+
         context->requireGLSLExtension("ARB_shader_stencil_export");
         name = "gl_FragStencilRef";
     }
     else if (semanticName == "sv_tessfactor")
     {
+        // TODO(JS): Need to ensure the adjustType can handle such a scenario. 
+        // float[2|3|4] in hlsl, float[4] on glsl (ie both are arrays but might be different size)
```

Comment:
Note also that this seems to be a case where HLSL nominally declares it as an array, but might allow a vector (or scalar) as well (I haven't looked into it).